### PR TITLE
Updates to wave_geometry

### PR DESCRIPTION
**Pull request by Kaythay**
I took the commits from the [update_install_scripts](https://github.com/wavelab/libwave/tree/update_install_scripts) branch that were related to `wave_geometry` and cherry-picked them to this branch. The original commits are:
https://github.com/wavelab/libwave/commit/b646f3f10215c4204a89da3fe3f00ba60a79800f
https://github.com/wavelab/libwave/commit/c4513c3e674a91a855891a6efaedfe1263921914
https://github.com/wavelab/libwave/commit/6e9d1eab275baffa3e2f0d2d2fd8236c9ca4fbfd
https://github.com/wavelab/libwave/commit/29fb128e04ef4b724fe05ab468b96aa1a9c8ee52

#### Pre-Merge Checklist:
- [ ] Code is documented in doxygen format
- [ ] Code has automated tests
- [ ] Zero compiler warnings
- [ ] Formatted with `clang-format`

### CODE REVIEW

File: wave_benchmark/include/wave/benchmark/pose_measurement.hpp [hunk 1]
```diff
@@ -16,11 +16,11 @@ namespace wave {
  *  @{ */
 
 struct BenchmarkPose {
-    Rotation rotation;
+    RotationMd rotation;
     Vec3 translation;
 
-    BenchmarkPose() : rotation(Rotation()), translation{0.0, 0.0, 0.0} {}
-    BenchmarkPose(Rotation rot, Vec3 trans)
+    BenchmarkPose() : rotation(RotationMd()), translation{0.0, 0.0, 0.0} {}
```
Comment (leokoppel) [inline wave_benchmark/include/wave/benchmark/pose_measurement.hpp]: Does this change behaviour? `RotationMd` is initialized to garbage, but old `Rotation` was initialized to identity IIRC

File: wave_benchmark/include/wave/benchmark/pose_measurement.hpp [hunk 2]
```diff
@@ -33,18 +33,20 @@ inline BenchmarkPose interpolate(const PoseMeasurement &m1,
                                  const TimePoint &t) {
     auto w2 = 1.0 * (t - m1.time_point) / (m2.time_point - m1.time_point);
 
-    auto m1inverse = m1.value.rotation;
-    m1inverse.invert();
+    auto m1inverse = inverse(m1.value.rotation);
 
-    auto relative = m2.value.rotation * m1inverse;
-    Vec3 wvec = w2 * relative.logMap();
+    auto relative = eval(log(m2.value.rotation * m1inverse));
```
Comment (leokoppel) [inline wave_benchmark/include/wave/benchmark/pose_measurement.hpp]: Can be replaced with `m2.value.rotation - m1.value.rotation` I think

log(a * inverse(b)) == a (box minus) b

File: wave_benchmark/include/wave/benchmark/pose_measurement.hpp [hunk 3]
```diff
@@ -16,11 +16,11 @@ namespace wave {
  *  @{ */
 
 struct BenchmarkPose {
-    Rotation rotation;
+    RotationMd rotation;
```
Comment (leokoppel) [inline wave_benchmark/include/wave/benchmark/pose_measurement.hpp]: I suspect there is not much point in `BenchmarkPose` anymore. If the point was to combine a Rotation and Translation, wave_geometry's `RigidTransformMd` does that.

As long as behaviour is unchanged by this PR, can add a new issue to make that change.

**Comment by Kaythay** [comment]
...I'm not really sure if I updated the submodule correctly.

**Comment by leokoppel** [comment]
Looks correct!

**Pull request merged**